### PR TITLE
Add support for Feature overrides to IsFeatureOverridden.

### PR DESCRIPTION
**Pull request by goodov**
Chromium uses `IsFeatureOverridden` in some places to enable/disable existing logic only if override is present. We should treat our brave-specific overrides in the same manner as they are external overrides, even though they're not external (non-griffin, non-cmdline, etc.). See [Blink-specific overrides](https://github.com/chromium/chromium/blob/8df70adc0091a6818772c55d68e1670a691f3a7d/content/child/runtime_features.cc#L137-L153) for example.

Resolves https://github.com/brave/brave-browser/issues/20436

Related https://github.com/brave/brave-browser/issues/18829

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: base/feature_override_unittest.cc [hunk 1]
```diff
@@ -21,20 +21,27 @@ const Feature kTestControlDisabledFeature{"TestControlDisabledFeature",
...
 const Feature kTestDisabledButOverridenFeature{
-    "TestDisabledButOverridenFeature", FEATURE_ENABLED_BY_DEFAULT};
+    "TestDisabledButOverridenFeature", FEATURE_DISABLED_BY_DEFAULT};
 
 constexpr Feature kTestConstexprEnabledButOverridenFeature{
-    "TestConstexprEnabledButOverridenFeature", FEATURE_DISABLED_BY_DEFAULT};
+    "TestConstexprEnabledButOverridenFeature", FEATURE_ENABLED_BY_DEFAULT};
 constexpr Feature kTestConstexprDisabledButOverridenFeature{
-    "TestConstexprDisabledButOverridenFeature", FEATURE_ENABLED_BY_DEFAULT};
+    "TestConstexprDisabledButOverridenFeature", FEATURE_DISABLED_BY_DEFAULT};
```
Comment (goodov) [inline base/feature_override_unittest.cc:31]: these feature were in a wrong state for the test, my bad. fixed it now.

File: chromium_src/content/child/runtime_features.cc [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-#define SetRuntimeFeaturesDefaultsAndUpdateFromArgs \
-  SetRuntimeFeaturesDefaultsAndUpdateFromArgs_ChromiumImpl
-#include "src/content/child/runtime_features.cc"
-#undef SetRuntimeFeaturesDefaultsAndUpdateFromArgs
-
-namespace content {
-
-void SetRuntimeFeaturesDefaultsAndUpdateFromArgs(
-    const base::CommandLine& command_line) {
-  SetRuntimeFeaturesDefaultsAndUpdateFromArgs_ChromiumImpl(command_line);
```
Comment (goodov) [inline chromium_src/content/child/runtime_features.cc:15]: nothing is going on here now. just a cleanup.

**Review by iefremov — APPROVED** [review]

**Comment by iefremov** [comment]
@mariospr please take a quick look

**Review by mariospr — APPROVED** [review]
LGTM!

**Pull request merged**